front_end/object_ui/CustomPreviewComponent.js

```diff
@@ -213,6 +213,37 @@ ObjectUI.CustomPreviewComponent = class {
     this.element = createElementWithClass('span', 'source-code');
     const shadowRoot = UI.createShadowRootWithCoreStyles(this.element, 'object_ui/customPreviewComponent.css');
     this.element.addEventListener('contextmenu', this._contextMenuEventFired.bind(this), false);
+
+    // DDT-specific operation to linkify an object's displayed runtime type
+    // Performed on all remote objects for which the following is resolvable:
+    // (Javascript) remoteObj["runtimeType"][Symbol(_type)]["[[FunctionLocation]]"]
+    // In Dart, this corresponds to an object's constructor's function location
+    let remoteFunction = "function remoteFunction() { " +
+        "return dart_library.debuggerLibraries()[0][\"dart\"].getReifiedType(this); }";
+    this._object._runtimeAgent.invoke_callFunctionOn({
+      objectId: this._object._objectId,
+      functionDeclaration: remoteFunction,
+      arguments: [],
+      silent: true,
+      returnByValue: false
+    }).then((remoteFunctionResult) => {
+      let remoteObject = this._object.runtimeModel().createRemoteObject(remoteFunctionResult.result);
+      remoteObject.debuggerModel().functionDetailsPromise(remoteObject).then((response) => {
+        if (response.location === undefined) return;
+        // Manually creating a link because formatting functions such as
+        // formatObjectAsFunction don't preserve the custom formatter's content.
+        let typeLink = createElementWithClass('span', 'linkified devtools-link');
+        typeLink.addEventListener('click', () => Common.Revealer.reveal(response.location) && false);
```

Comment:
Devtools code doesn't follow the Dart 80-column limit, but if it's easy, as in this case, let's format it to 80 columns anyway.